framework/platform/ios/ios_platform.h

```diff
@@ -33,6 +33,7 @@ class IosPlatform : public Platform
     void* view;
 
 protected:
+	bool uses_internal_looping() override { return false; }
```

Comment:
What does this `internal_looping` mean? Why is it different on IOS than everywhere else?

Reply:
On iOS, main_loop() is called on each frame by the platform display synchronization loop. See ViewController::renderLoop. This is generally how rendering is done on iOS.

On other platforms, it is called once. Code had recently been added to the top of `main_loop()` that assumed it would only be called once, which breaks on iOS.

Another option would be to separate the updating/rendering code from `main_loop()`, so it can be called separately on iOS. I decided a loop check would be a more minimal change, but I can easily take that approach, if it is a better factoring.

Reply:
ok... what do you think about having the one-time code in `main_loop()` and then calling into `main_loop_frame()` in a loop? Would that work?

Reply:
Good idea! I've implemented this now.